fdbserver/storageserver.actor.cpp

```diff
@@ -296,9 +298,58 @@ struct StorageServer {
 	std::map<Version, Standalone<VersionUpdateRef>> mutationLog; // versions (durableVersion, version]
 
 public:
+public:
+	// Histograms
+	struct FetchKeysHistograms {
+		const Reference<Histogram> latency;
+		const Reference<Histogram> bytes;
+		const Reference<Histogram> bandwidth;
+
+		FetchKeysHistograms(const Reference<Histogram> latency_, const Reference<Histogram> bytes_, const Reference<Histogram> bandwidth_)
+			: latency(latency_), bytes(bytes_), bandwidth(bandwidth_) {}
+	} fetchKeysHistograms;
+
+	class CurrentRunningFetchKeys {
+		std::unordered_map<UID, double> startTimeMap;
+		std::unordered_map<UID, KeyRangeRef> keyRangeMap;
+
+		static const StringRef emptyString;
+		static const KeyRangeRef emptyKeyRange;
+	public:
+		void recordStart(const UID id, const KeyRange keyRange) {
+			startTimeMap[id] =  now();
+			keyRangeMap[id] = keyRange;
+		}
+
+		void recordFinish(const UID id) {
+			startTimeMap.erase(id);
+			keyRangeMap.erase(id);
+		}
+
+		std::pair<double, KeyRangeRef> longestTime() const {
+			if (numRunning() == 0) {
+				return {-1, emptyKeyRange};
+			}
+
+			const double now_= now();
```

Comment:
This naming convention(`now_`), which is also used at other site, seems to be new in FDB code base. Maybe simply name it `now`?

Reply:
Sometimes, I use start and stop.  (now_ is strange, because it will be in the past by the time you use it, also, we don't usually end local variables with a `_`.